test/asyncio_tests/test_asyncio_client.py

```diff
@@ -214,6 +220,44 @@ async def test_list_database_names(self):
         self.assertIsInstance(names, list)
         self.assertIn(self.collection.database.name, names)
 
+    @unittest.skipIf(not contextvars, 'this test requires contextvars')
+    @asyncio_test
+    async def test_contextvars_support(self):
+        contextvar = contextvars.ContextVar('variable', default='default')
+
+        class Listener(monitoring.CommandListener):
+            def __init__(self):
+                self.vars = []
+
+            def save_contextvar_value(self):
+                self.vars.append(contextvar.get())
+
+            def started(self, event):
+                self.save_contextvar_value()
+
+            def succeeded(self, event):
+                self.save_contextvar_value()
+
+            def failed(self, event):
+                self.save_contextvar_value()
```

Comment:
I think we can just pass here since we don't expect a failure in this particular test?

Reply:
Done.